client/src/components/actionButtons.jsx

```diff
@@ -35,19 +38,82 @@ const SideButton = styled.a`
   color: #666;
   background-color: #f7f7f7;
   font-weight: bold;
-  font-size: .9rem;
+  font-size: .7rem;
+  display: inline;
+  &:hover{
+    background: #fffff1;
+  }
+`
+const WriteAReviewLink = styled.a`
+  text-decoration: none;
+  color: white;
+`
+const Link = styled.a`
+  color: #666;
+  text-decoration: none;
 `
 
-const ActionButtons = (props) => {
+class ActionButtons extends React.Component {
+  constructor(props){
+    super(props)
+    
+    this.state = {
+      isModal: false, // should default to false
+      modalState: "" // should default to "" 
+    }
+    this.toggleModal = this.toggleModal.bind(this);
+    }
 
-  return (
-    <ActionButtonsDiv>
-      <WriteAReview><Star className="fas fa-star" /> Write A Review</WriteAReview>
-      <SideButton><i className="fas fa-camera"></i>  Add Photo</SideButton>
-      <SideButton><i class="fas fa-share-square"></i>  Share</SideButton>
-      <SideButton><i class="fas fa-bookmark"></i>  Save</SideButton>
+  toggleModal(modalType){
+    this.setState({
+      isModal: !this.state.isModal,
+      modalState: modalType
+    })
+    console.log(this.state)
+  }
+    
+    render(){
+      // conditional rendering of modal content state
+      let modalContentState
+      
+      if (this.state.modalState === "share") {
+        modalContentState = <ShareModal toggleModal={this.toggleModal}/>
+      } else if (this.state.modalState === "save"){
+        modalContentState = <ModalContent>
+          <div>save</div>
+          <div onClick={() => this.toggleModal(null)}>Close</div>
+        </ModalContent>
+      }
+
+
+      return (
+        <ActionButtonsDiv>
+          <Modal onClick={() => this.toggleModal(null)} isModal={this.state.isModal}>
+          </Modal>
+            {modalContentState}
+
+        <WriteAReview><Star className="fas fa-star" /><WriteAReviewLink href="#">Write A Review</ WriteAReviewLink></WriteAReview>
+        <SideButton>
+          <Link href="#">
+          <i className="fas fa-camera"></i>  Add Photo
+          </Link>
+        </SideButton>
+
+        <SideButton onClick={() => this.toggleModal("share")}>
```

Comment:
yep i'll have to install the linter soon. I have a bunch of double quotes everywhere... this probably wont get fixed before this pull request but it will be done eventually (as I found more double quotes)